apiserver/facades/agent/uniter/service.go

```diff
@@ -73,6 +74,23 @@ type ApplicationService interface {
 
 	// DestroyUnit prepares a unit for removal from the model.
 	DestroyUnit(ctx context.Context, unitName coreunit.Name) error
+
+	// WatchApplication returns a NotifyWatcher for changes to the application.
+	WatchApplication(ctx context.Context, name string) (watcher.NotifyWatcher, error)
+
+	// GetApplicationIDByUnitName returns the application ID for the named unit.
+	//
+	// Returns [applicationerrors.UnitNotFound] if the unit is not found
```

Comment:
**todo:** This short hand won't back link to any doc, since this package is not imported. You'll need to give the fully qualified name.

(same for the comment below)